feat/feature.go

```diff
@@ -105,6 +105,42 @@ type Collection interface {
 	Location() Feature
 }
 
+// BasePositionOf returns the position in f converted to coordinates in the
+// deepest feature location and the deepest non nil reference feature, which
+// may be the feature itself if it has a nil Location.
+// BasePositionOf will panic if the feature chain is deeper than 1000 links.
+func BasePositionOf(f Feature, position int) (int, Feature) {
+	for n := 0; n < 1000; n++ {
+		position += f.Start()
+		if f.Location() != nil {
+			f = f.Location()
+			continue
+		}
+		return position, f
+	}
+	panic("feat: feature chain too long")
+}
+
+// PositionWithin returns the position in f converted to coordinates in a
+// given reference feature and a boolean value indicating if the conversion
```

Comment:
"... and a boolean indicating whether f can be located relative to ref." and drop the following sentence.